doc/experiment.rst

```diff
@@ -1,327 +1,440 @@
-ensure that prefixes are used with models and scripts.
-
-We modify the example above to enable key prefixing as follows:
-
-.. code-block:: bash
-
-  import numpy as np
-  from smartsim import Experiment
-
-  exp = Experiment("Training-Run", launcher="slurm")
-
-  # setup ensemble parameter space
-  learning_rate = list(np.linspace(.01, .5))
-  train_params = {"LR": learning_rate}
+``Ensemble`` will create the underlying ``Model`` objects. Users may also
+manually create and append ``Model(s)`` to an ``Ensemble``.
+Lastly, the :ref:`Ensemble API<ensemble_api>` supports launching Machine Learning Models, TensorFlow
+scripts and functions at runtime to enable AI and ML within an Ensemble
+Workload.
+
+Ensemble Prefixing
+------------------
+If each of the ensemble members attempt to use the
+same code to access their respective data in the Orchestrator,
+the names used to reference data, models, and scripts will be identical,
+and without the use of SmartSim and SmartRedis helper methods, ensemble members
+will end up inadvertently accessing or overwriting each other’s data. To prevent
+this situation, the SmartSim ``Ensemble`` object supports
+key prefixing, which automatically prepends the name
+of the model to the keys by which it is accessed. With
+this enabled, collision is resolved and
+ensemble members can use the same code.
+
+For example, assume you have two models in the ``Ensemble`` object,
+named `bar_0` and `bar_1`. In the application code you
+use the function ``Client.put_tensor("foo")``. With
+ensemble key prefixing turned on, the `bar_0` and `bar_1` Model applications
+can access the tensor `"foo"` by name without overwriting or accessing the other
+ensemble member's `"foo"` tensor.
+
+Create a Ensemble
+-----------------
+An ``Ensemble`` is created through the factory method: ``Experiment.create_ensemble()``.
+The ``create_ensemble()`` factory method returns an initialized ``Ensemble`` object that
+gives you access to functions associated with the :ref:`Ensemble API<ensemble_api>`.
+To initialize an Ensemble, a user must follow one of the three methods of ensemble
+creation:
+
+1. Manual Model Appending
+     A technique that allows users to create and add model instances to an ensemble, offering a level
+     of customization in ensemble design.
+2. Parameter Expansion
+     A technique that allows users to set parameter values and control how the parameter values
+     spread across the ensemble members by specifying a permutation strategy.
+3. The Utilization of Replicas
+     A technique that allows users to create identical or closely related models within an ensemble. Users can assess
+     how a system responds to the same set of parameters under multiple instances.
+
+.. note::
+  For more information and instruction on ensemble creation methods, navigate to the :ref:`Ensemble documentation page<dead_link>`.
+
+==================
+Experiment Example
```

Comment:
Should we just call it Example since it is under the Experiment heading already?

Reply:
yes, the other sections follow this format